src/Chapter3.hs

```diff
@@ -971,6 +1076,31 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data WeekDay = Sunday 
+    | Monday 
+    | Tuesday 
+    | Wednesday
+    | Thursday 
+    | Friday 
+    | Saturday
+    deriving (Show, Enum)
+
+isWeekend :: WeekDay -> Bool
+isWeekend day = case day of
+    Saturday -> True
+    Sunday -> True
+    _ -> False
+
+nextDay :: WeekDay -> WeekDay
+nextDay day = case day of
+    Saturday -> Sunday      -- Is there a way in which I can avoid a special case for "Saturday" ??
```

Comment:
That is right!
You could also use the instance of `Bounded` in here:

```suggestion
    | day == maxBound = minBound
```

This is also a bit more maintainable. If you decide, for example, to have the "European" system of week days in the data type constructors, you won't need to change this function at all! 🙂